src/interchange/src/avro.rs

```diff
@@ -1149,23 +1149,38 @@ struct DebeziumDeduplicationState {
     /// Messages that are not ahead of the last recorded pos/row will be skipped.
     binlog_offsets: HashMap<String, (usize, usize, Option<i64>)>,
     /// Whether or not to track every message we've ever seen
-    full: TrackFull,
+    full: Option<TrackFull>,
 }
 
 /// If we need to deal debezium possibly going back after it hasn't seen things
 #[derive(Debug)]
-enum TrackFull {
-    No,
-    /// Track all offsets that have ever been seen
+struct TrackFull {
     /// binlog filename to offset
-    SeenOffsets(HashMap<String, HashSet<(usize, usize)>>),
+    seen_offsets: HashMap<String, HashSet<(usize, usize)>>,
+    seen_snapshot_keys: HashMap<String, HashSet<Row>>,
+    key_indices: Option<Vec<usize>>,
+    key_buf: RowPacker,
+}
```

Comment:
I would really love a comment high-level overview of the problems that this struct is trying to solve, either here or by amending the big comment on `DebeziumDeduplicationStrategy`.